### PR TITLE
Ability to publish package from branch for development and test purpose

### PR DESCRIPTION
**Pull request by naohiro-t**
Package with this github action will look like `2.7.2-dev.3` (`{version}-dev.{github_action_number}`)

### CODE REVIEW

**Review by matthewchigira — CHANGES_REQUESTED** [review]

File: .github/bin/publish-package.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+if [ -n "${GITHUB_HEAD_REF}" ]; then
+  export BRANCH="${GITHUB_HEAD_REF#refs/heads/}"
+else
+  export BRANCH="${GITHUB_REF#refs/heads/}"
+fi
+
+export BUILD_NUM=${GITHUB_RUN_NUMBER}
+export tag=$(build-utils publishprepare --branch "${BRANCH}" --build-num "${BUILD_NUM}")
+echo "${tag}"
+npm publish --tag "${tag}" --verbose || true
```
[the remark this answers is not in the document]
Comment (naohiro-t) [reply .github/bin/publish-package.sh:14]: We can make always make it "dev" too! Just thought branch name is more clear which package you want :) 

> The name of the package comes from the name specified in the package.json file.
Yes, `build-utils publishprepare` will update package.json, then publish.

[the remark this answers is not in the document]
Comment (naohiro-t) [reply .github/bin/publish-package.sh:14]: Changed tag to be always "dev".

[the remark this answers is not in the document]
Comment (matthewchigira) [reply .github/bin/publish-package.sh:14]: Sorry for opening this again (:smile: )

Don't we just want to use `build-utils` to write the update into `package.json` file and then just publish with the tag of `dev`?

```
build-utils publishprepare --branch dev --build-num "${BUILD_NUM}"
npm publish --tag dev --verbose || true
```

[the remark this answers is not in the document]
Comment (naohiro-t) [reply .github/bin/publish-package.sh]: My change does the same as what you described, but your way is cleaner :) So, I've updated!

[the remark this answers is not in the document]
Comment (matthewchigira) [reply .github/bin/publish-package.sh]: Thanks! :sunglasses:

**Review by matthewchigira — APPROVED** [review]
LGTM :rocket:

**Pull request merged**